libs/feature/search/src/lib/favorites/favorite-star/favorite-star.component.spec.ts

```diff
@@ -69,6 +85,22 @@ describe('FavoriteStarComponent', () => {
     expect(component).toBeTruthy()
   })
 
+  describe('default login URL', () => {
+    it('should construct a localised GN login URL by default', () => {
+      expect(component.loginUrl).toEqual(
+        '/geonetwork/srv/fre/catalog.signin?redirect=http://localhost/'
+      )
+    })
+  })
+  describe('config login URL', () => {
+    beforeEach(() => {
+      loginUrl = '/cas/login?service='
```

Comment:
This test doesn't pass yet, since I did not manage to change the value of the `InjectionToken` from one test to the next. Would greatly appreciate any help on this @jahow @fgravin :pray:

Reply:
Yes, this is what I tried to do. Somehow, I don't manage to change the value returned by the factory.

Reply:
Yes it's because the providers are managed once.
See https://stackoverflow.com/questions/58630884/override-a-testbed-provider-for-specific-test

Reply:
Thanks, I stumbled over this as well :-) So, if understand correctly, once `.compileComponents()` has been called the values are frozen, even when using a factory. I've reorganized the test suite a bit to override the provider before.